12_glob_lab/globlab.txt

```diff
@@ -5,25 +5,44 @@ https://www.tocode.co.il/bundles/unix1/lessons/globlab
 
   1. All files whose name starts with a lowercase letter
   Answer: 
+  
+	ls -d [a-z]*
 
   2. All files whose name starts with an uppercase letter
   Answer:
 
+	ls -d [A-Z]*
+  
   3. All files whose name ends in an "a", "e", "i", "o" or "u"
   Answer:
 
+	ls -d *[a,e,i,o,u]
+  
   4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
   Answer:
 
+	ls -d *[!a,e,i,o,u]
+  
   5. All files with extension ".c" that are in a folder named "src"
   Answer:
+	
+	ls ./src/*.c
 
   6. All files with extension ".c" under any folder
   Answer:
 
+    find . -name '*.c'
+	ls -R | grep '.*[.]c'
```

Comment:
wow nice :) really liked your solution! here are a few notes:

this command ``` ls -R | grep '.*[.]c ```  
will also match the file ```fake.c.other```
to fix this add a ```$``` to the end: ``` ls -R | 'grep '.*[.]c$'``` 
(this is a regex thing...)

also there is a cool and almost  hidden feature of bash (4 and up) called "globstar" that allows you to create recursive globs, like this:
``` ls **/*.c # will match any .c file under current dir```

just remember that you need to enable it first: ``` shopt -s globstar ```. 

it's good  to know about it because newer shells like zsh have it enabled by default, and many build tools use it (this is how i found out about it)